main.cpp

```diff
@@ -231,214 +333,320 @@ struct BasketballCourt
+}
+
+void AudioDepartment::testAudioSoftware()
+{
+    bool softwareWorking = true;
+    if(softwareWorking == true)
+    {
+        std::cout << "anounce sales department";
+    }
+    else
+    {
+        std::cout << "keep working on the software";
+    }
+}
+
+void AudioDepartment::recordAudio(int numberOfMicrophones)
+{
+    std::cout << "record" << numberOfMicrophones << "different sources simultaneously";
+}
 
- */
 struct VisualDepartment
 {
-    //5 properties:
-    //    1) number of videocameras (int)
     int numberOfVideocameras = 11;
-    //    2) number of light projectors (int)
     int numberOfLightProjectors = 8;
-    //    3) department logo (std::string)
     std::string departmentLogo = "all Video";
-    //    4) prefered editing software (std::string)
     std::string preferedEditingSoftware = "davinci resolve";
-    //    5) prefered cloud storage solution (std::string)
     std::string preferedCloudStorageSolution = "dropbox for business";
-    //3 things it can do:
-    //    1) create animation
+
     void createAnimation();
-    //    2) color grade video footage
     void colorGradeVideoFootage();
-    //    3) edit short films
     void editShortFilms();
-
 };
 
-/*
-Thing 7) sales
+void VisualDepartment::createAnimation()
+{
+    int numberOfAnimators = 12;
+    std::cout << "employ" << numberOfAnimators << "people to create the animation";
+}
+
+void VisualDepartment::colorGradeVideoFootage()
+{
+    int numberOfColorsAvailable = 1231232;
+    std::string colorProfile = "RGB";
+    std::cout << "use the" << numberOfColorsAvailable << "available colours in the" << colorProfile << "profile";
+}
+
+void VisualDepartment::editShortFilms()
+{
+    float filmLength = 32.42f;
+    float rawFootageLength = 123.51f;
+    std::cout << "edit the" << rawFootageLength << "minutes into a" << filmLength << "minutes long film";
+}
 
- */
 struct Sales
 {
-    //5 properties:
-    //    1) software copies sold (int)
     int softwareCopiesSold = 19842;
-    //    2) software price (float)
     float softwarePrice = 98.12f;
-    //    3) number of steady clients (int)
     int numberOfSteadyClients = 84;
-    //    4) number of subscribers to cloud solutions (int)
     int numberOfSubscribersToCloudSolutions = 81;
-    //    5) department email address (std::string)
     std::string departmentEmailAddress = "companymail@gmail.com";
-    //3 things it can do:
-    //    1) sell software
-    float sellSoftware(float softwarePrice); //returns the amount of money earned
-    //    2) reach out to potential clients
+
+    float sellSoftware(float softwarePrice);
     void reachOutToPotentialClients();
-    //    3) negotiate better deals for large contracts
     void negotiateBetterDealsForLargeContracts();
-
 };
 
-/*
-Thing 8) RnD department
+float Sales::sellSoftware(float softwareCopyPrice)
+{
+    int numberOfSoftwareCopiesSold = 144;
+    float copiesFloat;
+    copiesFloat = float(numberOfSoftwareCopiesSold);
```

Comment:
Casting has not been covered, and this is a C-Style cast.
remove it
find another way to solve the warnings related to this type conversion.
- [ ] revise